src/utils/issue.ts

```diff
@@ -0,0 +1,200 @@
+    .get({
+      owner: owner || payload.repository.owner.login,
+      repo: repo || payload.repository.name,
+      issue_number: issueNum || payload.issue.number,
+    })
+    .then(({ data }) => data as Issue);
+}
+
+export async function fetchIssueComments(params: FetchParams) {
+  const { octokit, payload } = params.context;
+  const { issueNum, owner, repo } = params;
+
+  return await octokit
+    .paginate(octokit.issues.listComments, {
+      owner: owner || payload.repository.owner.login,
+      repo: repo || payload.repository.name,
+      issue_number: issueNum || payload.issue.number,
+    })
+    .then((comments) => comments as IssueComments);
+}
+
+export async function fetchLinkedIssues(params: FetchParams, comments?: IssueComments) {
+  let issueComments: IssueComments | undefined = comments;
+  const linkedIssues: {
+    issueNumber: number;
+    repo: string;
+  }[] = [];
+
+  if (!issueComments && !params) {
+    throw new Error("Either issueComments or params must be provided");
+  }
+
+  if (!issueComments) {
+    issueComments = await fetchIssueComments(params);
+  }
+
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  if (!issueComments) {
+    logger.info("No comments found on issue");
+    return linkedIssues;
+  }
+
+  for (const comment of issueComments) {
+    const linkedIssue = idIssueFromComment(login, comment.body);
+    if (linkedIssue) {
+      linkedIssues.push(linkedIssue);
+    }
+  }
+
+  return await filterLinkedIssues(params, linkedIssues);
+}
+
+async function filterLinkedIssues(params: FetchParams, linkedIssues: { issueNumber: number; repo: string }[]) {
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  const contextIssues: {
+    issueNumber: number;
+    repo: string;
+  }[] = [];
+
+  for (const issue of linkedIssues) {
+    if (issue && issue.issueNumber && issue.repo) {
+      if (await isRepoFromSameOrg(params.context, issue.repo, login)) {
+        contextIssues.push({
+          issueNumber: issue.issueNumber,
+          repo: issue.repo,
+        });
+      } else {
+        logger.info(`Ignoring linked issue ${issue.issueNumber} from ${issue.repo} as it is not from the same org`);
+      }
+    }
+  }
+
+  return contextIssues;
+}
+
+export async function getLinkedIssueContextFromComments(context: Context, issueComments: IssueComments) {
+  // find any linked issues in comments by parsing the comments and enforcing that the
+  // linked issue is from the same org that the current issue is from
+  const linkedIssues = await fetchLinkedIssues({ context }, issueComments);
+
+  // the conversational history of the linked issues
+  const linkedIssueComments: IssueComments = [];
+
+  // we are only going one level deep with the linked issue context fetching
```

Comment:
Why not all? Context length is not a problem anymore.

Reply:
While we have a 128k window I don't think it's going to be as effective stuffing one prompt/chat history with near that amount of tokens. If we are adding more context which could potentially be 30/40k worth of tokens GPT will get lost in the sauce with it a little I think.

Also typically all issues are going to have at least one other linked issue which could easily produce deep enough recursion to max our tokens. Also that GPT output has to be factored into this 128k (likely always < 10k I'd imagine).

<hr/>

One deep seems ideal to me, it obtains strictly the most relevant context without adding too much noise but if not one level deep I'd suggest to cap it somehow whether through a depth value or token limit

Reply:
I prefer to try utilizing the most amount of context. We can make assumed optimizations later.